src/header.rs

```diff
@@ -1614,15 +1614,34 @@ pub fn bytes2path(bytes: Cow<[u8]>) -> io::Result<Cow<Path>> {
 pub fn bytes2path(bytes: Cow<[u8]>) -> io::Result<Cow<Path>> {
     Ok(match bytes {
         Cow::Borrowed(bytes) => {
-            Cow::Borrowed({ Path::new(str::from_utf8(bytes).map_err(invalid_utf8)?) })
-        }
-        Cow::Owned(bytes) => {
-            Cow::Owned({ PathBuf::from(String::from_utf8(bytes).map_err(invalid_utf8)?) })
+            Cow::Borrowed(Path::new(str::from_utf8(bytes).map_err(invalid_utf8)?))
         }
+        Cow::Owned(bytes) => Cow::Owned(PathBuf::from(
+            String::from_utf8(bytes).map_err(invalid_utf8)?,
+        )),
     })
 }
 
 #[cfg(target_arch = "wasm32")]
 fn invalid_utf8<T>(_: T) -> io::Error {
     io::Error::new(io::ErrorKind::InvalidData, "Invalid utf-8")
 }
+
+pub fn canonicalize(path: &Path) -> PathBuf {
+    canonicalize_or_err(path).unwrap_or(path.to_path_buf())
+}
+
+pub fn canonicalize_or_err(path: &Path) -> io::Result<PathBuf> {
+    if cfg!(target_os = "wasi") {
+        // canonicalize is not yet supported in the standard library
+        // see: https://github.com/rust-lang/rust/blob/master/library/std/src/sys/wasi/fs.rs#L609
```

Comment:
I think this comment can be refined to how the wasi platform doesn't have a canonicalize method right now.

Reply:
Not sure what you mean - I linked to the exact line of code that isn't implemented.
How can I refine that more :-)
Cheers